src/modules/IscsiClientLib.rb

```diff
@@ -418,17 +418,28 @@ def saveConfig(user_in, pass_in, user_out, pass_out)
       tmp_conf = deep_copy(@config)
       tmp_val = Ops.get_list(tmp_conf, "value", [])
 
-      if Ops.greater_than(Builtins.size(user_in), 0) &&
-          Ops.greater_than(Builtins.size(pass_in), 0)
-        tmp_val = setOrAdd(tmp_val, "node.session.auth.username", user_in)
-        tmp_val = setOrAdd(tmp_val, "node.session.auth.password", pass_in)
+      if !user_in.empty? && !pass_in.empty?
+        tmp_val = setOrAdd(
+          tmp_val,
+          "discovery.sendtargets.auth.authmethod",
+          "CHAP"
+        )
+        tmp_val = setOrAdd(
+          tmp_val,
+          "discovery.sendtargets.auth.username_in",
+          user_in
+        )
+        tmp_val = setOrAdd(
+          tmp_val,
+          "discovery.sendtargets.auth.password_in",
+          pass_in
+        )
       else
-        tmp_val = delete(tmp_val, "node.session.auth.username")
-        tmp_val = delete(tmp_val, "node.session.auth.password")
+        tmp_val = delete(tmp_val, "discovery.sendtargets.auth.username_in")
```

Comment:
Both, user and password, have to be specified (for incoming and outgoing authentication).
In fact the check is already done in widgets.rb before calling saveConfig (the values are reset in case only one is specified).
I have added "specified" for better readability.
Thanks for review.